deepset_cloud_sdk/workflows/sync_client/files.py

```diff
@@ -108,3 +112,45 @@ def upload_texts(
             timeout_s=timeout_s,
         )
     )
+
+
+def list_files(
+    api_key: Optional[str] = None,
+    api_url: Optional[str] = None,
+    workspace_name: str = DEFAULT_WORKSPACE_NAME,
+    name: Optional[str] = None,
+    content: Optional[str] = None,
+    filter: Optional[str] = None,
+    batch_size: int = 100,
+    timeout_s: int = 300,
+) -> List[File]:
+    """List files in the Deepset Cloud.
+
+    WARNING: this will only work for workspaces up to 1000 files.
+    TODO: make this a generator
+
+    :param api_key: API key to use for authentication.
+    :param api_url: API URL to use for authentication.
+    :param workspace_name: Name of the workspace to list the files from.
+    :param name: Name of the file to filter for.
+    :param content: Content of the file to filter for.
+    :param filter: Filter to apply to the file list.
+    :param batch_size: Batch size to use for the file list.
+    """
+
+    async def list_files_async() -> List[File]:
```

Comment:
Assuming we can't just wrap the function defined in the async workflow because of the async generator?

Reply:
Yes. You would need some python magic here so I thought for now we just list it with a timeout and suggest to use the async implementation. Lets fix this once we really need this feature, since i just added this to test the sdk structure

Reply:
I think this is probably a better solution than magic tbh 😅

Reply:
lets see :)